src/mscorlib/src/System/Globalization/EncodingTable.Unix.cs

```diff
@@ -1,101 +1,162 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Text;
+
 namespace System.Globalization
 {
-    using System;
-    using System.Text;
-    using System.Collections;
-    using System.Collections.Generic;
-    using System.Runtime.CompilerServices;
-    using System.Runtime.InteropServices;
-    using System.Runtime.Versioning;
-    using System.Security;
-    using System.Threading;
-    using System.Diagnostics.Contracts;
-
     internal static class EncodingTable
     {
+        //
+        // This caches the data item which is indexed by the code page value.
```

Comment:
It's probably worth commenting on why this is using a Hashtable instead of a Dictionary.  I assume it's because Hashtable supports readers concurrent with a single writer, which means reading from a Synchronized Hashtable doesn't require locking on reads, whereas protecting a Dictionary would?  What about ConcurrentDictionary?  Hashtable is fine for now... at some point it might be worthwhile examining whether a switch could be better.